src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int SHAPE_RANDOM_BOUND = FigureEnum.values().length;
+    private static final int VALUE_RANDOM_BOUND = 20;
+    private static final String DEFAULT_COLOR = Color.WHITE.name().toLowerCase();
+    private static final int DEFAULT_VALUE = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        FigureEnum figure = FigureEnum.values()[random.nextInt(SHAPE_RANDOM_BOUND)];
+        String color = colorSupplier.getRandomColor();
+        int valueFirst = random.nextInt(VALUE_RANDOM_BOUND);
+        int valueSecond = random.nextInt(VALUE_RANDOM_BOUND);
+        int valueThird = random.nextInt(VALUE_RANDOM_BOUND);
+        switch (figure) {
+            case CIRCLE: {
+                return new Circle(color, valueFirst);
+            }
+            case ISOSCELES_TRAPEZOID: {
+                return new IsoscelesTrapezoid(color, valueFirst, valueSecond, valueThird);
+            }
+            case RECTANGLE: {
+                return new Rectangle(color, valueFirst, valueSecond);
+            }
+            case RIGHT_TRIANGLE: {
+                return new RightTriangle(color, valueFirst, valueSecond);
+            }
+            default: {
+                return new Square(color, valueFirst);
+            }
+        }
```

Comment:
Don't return `null` from a method. In the `switch` statement, you have covered all possible cases of the `FigureEnum`. Therefore, you do not need the `default` case to return a `Square`. You can explicitly list the `SQUARE` case and remove the `default` case.